dev/src/index.ts

```diff
@@ -502,30 +481,55 @@ export class Firestore implements firestore.Firestore {
       this._projectId = settings.projectId;
     }
 
-    if (settings.host !== undefined) {
+    let url: URL | null = null;
+
+    // If the environment variable is set, it should always take precedence
+    // over any user passed in settings.
+    if (process.env.FIRESTORE_EMULATOR_HOST) {
+      validateHost(
+        'FIRESTORE_EMULATOR_HOST',
+        process.env.FIRESTORE_EMULATOR_HOST
+      );
+
+      settings = {
+        ...settings,
+        host: process.env.FIRESTORE_EMULATOR_HOST,
+        ssl: false,
+      };
+      url = new URL(`http://${settings.host}`);
+    } else if (settings.host !== undefined) {
       validateHost('settings.host', settings.host);
-      if (settings.servicePath !== undefined) {
-        throw new Error(
-          'Cannot set both "settings.host" and "settings.servicePath".'
-        );
-      }
-      if (settings.apiEndpoint !== undefined) {
-        throw new Error(
-          'Cannot set both "settings.host" and "settings.apiEndpoint".'
+      url = new URL(`http://${settings.host}`);
+      if (
+        (settings.servicePath !== undefined &&
+          settings.servicePath !== url.hostname) ||
+        (settings.apiEndpoint !== undefined &&
+          settings.apiEndpoint !== url.hostname)
+      ) {
+        // eslint-disable-next-line no-console
+        console.warn(
```

Comment:
I am debating if it makes sense to log this warning even when FIRESTORE_EMULATOR_HOST is used. Most of the time, this shouldn't warn since servicePath and apiEndpoint are rarely set.

Reply:
Yeah, I think that makes more sense. If anything, it will make it easier for developers to figure out what host is actually being used.